src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc

```diff
@@ -0,0 +1,484 @@
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::REJECTED;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  VoiceCommunicationAppSetup();
+
+  RunCommandExpectations(
+      msg_mobile_response, mobile_info, mobile_response, success, command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_IndicatorWaitingForResponse_IGNORED) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::IGNORED;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  MediaAppSetup();
+
+  ON_CALL(*mock_app_, AddIndicatorWaitForResponse(_))
+      .WillByDefault(Return(false));
+
+  RunCommandExpectations(
+      msg_mobile_response, mobile_info, mobile_response, success, command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       Run_SetAudioStreamingIndicatorRequestOK_SendToHmi) {
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MediaAppSetup();
+  ON_CALL(*mock_app_, AddIndicatorWaitForResponse(_))
+      .WillByDefault(Return(true));
+
+  DefineHMILevelUIAvailable();
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_));
+
+  command->Run();
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest, onTimeOut_GENERIC_ERROR) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  (*msg_mobile_response)[am::strings::msg_params][am::strings::result_code] =
+      mobile_apis::Result::GENERIC_ERROR;
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::GENERIC_ERROR;
+  const bool success = false;
+  const char* mobile_info = "";
+
+  EXPECT_CALL(
+      mock_message_helper_,
+      CreateNegativeResponse(_, _, _, am::mobile_api::Result::GENERIC_ERROR))
+      .WillOnce(Return(msg_mobile_response));
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&msg_mobile_response), Return(true)));
+
+  command->onTimeOut();
+
+  ResultCommandExpectations(
+      msg_mobile_response, mobile_info, mobile_response, success);
+}
```

Comment:
@vlantonov 
Additional test case is added
Done in 1456832e723cd462c70ff1e55e4818cce6867ef7